Assets/MixedRealityToolkit.SDK/Experimental/NoneNativeKeyboard/Scripts/UICollection.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// This component represents and ordered collection of UI elements. You
+    /// can add to the UICollection by either dropping UI elements in the UI
+    /// element this component is attached to in the Editor or by calling the
+    /// AddItem(RectTransform) method. UI elements that are added to this
+    /// collection via the Editor will be automatically arranged when this
+    /// component executes.
+    /// To use this component attach it to a UI element (a GameObject with a
+    /// RectTransform component) such as an Image or Panel.
+    /// </summary>
+    [AddComponentMenu("UI/HoloUIKit/UICollection")]
+    [RequireComponent(typeof(RectTransform))]
+    [ExecuteInEditMode]
+    public class UICollection : MonoBehaviour
```

Comment:
Interesting, curious what features of TileGridCollection or GridObjectCollection were missing that required you to use this?

Reply:
Since this was a straight port I just kept it with the initital keyboard. I tried replacing it with either GridObjectCollection or TileGridObjectCollection, but at least two problems occured:
- No matter the choice of column or row first, the gameobjects get sorted column first
- The width of the tiles is not taken into account. the Enter key has twice the width and gets overlapped by the next tile

This component could actually be a valuable addition, because you give it a square and spacings and it fits its items in based on their size.

Reply:
Ah, good point regarding the width of tiles! By the way, fixed "no matter the choice of column or row first, the gameobjects get sorted column first" in #6550 so the grid object collection layout should work better in 2.2